src/lm_buddy/jobs/_entrypoints/prometheus.py

```diff
@@ -0,0 +1,187 @@
+            print(
+                f"[w] {e.message}, "
+                f"retrying ({current_retry_attempt}/{config.evaluation.max_retries})"
+            )
+            current_retry_attempt += 1
+            if current_retry_attempt > config.evaluation.max_retries:
+                raise e
+    return (feedback, score)
+
+
+def run_eval(
+    config: PrometheusJobConfig, data: Dataset, tokenizer: PreTrainedTokenizer, client: OpenAI
+) -> str:
+    # enable / disable tqdm
+    dataset_iterable = tqdm(data) if config.evaluation.enable_tqdm else data
+
+    # open the output file for writing and iterate on samples
+    tracking_name = config.tracking.name if config.tracking is not None else "output.json"
+    output_fname = Path(config.evaluation.output_folder) / tracking_name
+    with output_fname.open("w") as file:
+        for sample in dataset_iterable:
+            # convert instructions from the dataset (`text_field` in a dict) to
+            # prompts that prometheus accepts
+            prompt = instruction_to_prompt(config, sample[config.dataset.text_field])
+
+            # skip those examples which are too long
+            tokenized_prompt = tokenizer(prompt, truncation=False)
+            if len(tokenized_prompt["input_ids"]) > 3072:
+                continue
+
+            # prepare output
+            result = copy.deepcopy(sample)
+            result["prometheus_output"] = []
+            result["prometheus_score"] = []
+
+            for idx in range(config.evaluation.num_answers):
+                (feedback, score) = get_response_with_retries(
+                    config, client, prompt, config.evaluation.max_retries
+                )
+                result["prometheus_output"].append(feedback)
+                result["prometheus_score"].append(score)
+
+            # dump sample results incrementally
+            file.write(json.dumps(result) + "\n")
+
+    # convert plain json dataset in HF format
+    output_hf_name = str(Path(config.evaluation.output_folder) / "hf" / tracking_name)
+    ds = load_dataset("json", data_files=str(output_fname), split="train")
+    ds.save_to_disk(output_hf_name)
+
+    return str(output_hf_name)
+
+
+def run_prometheus(config: PrometheusJobConfig, artifact_loader: ArtifactLoader):
+    # instantiate OpenAI client to speak with the vLLM endpoint
+    client = OpenAI(base_url=config.prometheus.inference.base_url)
+
+    # Register a dataset file artifact if tracking is enabled
+    if config.tracking:
+        with wandb_init_from_config(config.tracking, job_type=LMBuddyJobType.EVALUATION):
+            # load dataset from W&B artifact
+            hf_loader = HuggingFaceAssetLoader(artifact_loader)
+            data = hf_loader.load_dataset(config.dataset)
+
+            # get the tokenizer
+            tokenizer_config = AutoTokenizerConfig(load_from=config.prometheus.inference.engine)
+            tokenizer = hf_loader.load_pretrained_tokenizer(tokenizer_config)
```

Comment:
Moving this into the `run_eval` method would reduce some code duplication, although not a huge issue.